File: SkillSurvey.SiteAdapters/Seek.js
const bassClass = require("./SiteAdapterBaseClass.js");

class Seek extends bassClass.SiteAdapterBaseClass {
    constructor(parameters) {
        super(parameters);
        this.configSettings = require("./Seek.json");
        this.siteName = "Seek.com.au";

        this.jobPostLink = '[data-automation="jobTitle"]';

        this.titleSelector = '[data-automation="job-detail-title"]';
        this.bodySelector = '[data-automation="jobAdDetails"]';
        this.postedDateSelector = "span.FYwKg._2Bz3E.C6ZIU_4._6ufcS_4._3KSG8_4._29m7__4._2WTa0_4";
        this.citySelector = "div.FYwKg._3VxpE_4 > div:nth-child(1)";
        this.country = "Australia";
        this.suburbSelector = "div.FYwKg._3VxpE_4 > div:nth-child(2)";

        this.titleType = this.bodyType = this.postedDateType = this.cityType = this.suburbType = "text";
    }
    
    GetJobSiteNumber(url, doc) {
        return url.substring(url.lastIndexOf("/job/") + 5, url.lastIndexOf("?"));
    }

    GetPostedDate (url, doc) {
        var postedDate = new Date();

        var ageString = doc.$(this.postedDateSelector).eq(0).text().replace("Posted ", "");
        var daysOld = 0;
        var daysIndex = ageString.indexOf("d ago");

        if (daysIndex !== -1) {
            // If "h ago", the advert is 0 days old, leave daysOld as 0
            daysOld = parseInt(ageString.substring(0, daysIndex));
        }

        // setDate changes postedDate to next month if argument is more then number of days in
        // current month
        postedDate.setDate((new Date()).getDate() - daysOld);

        return postedDate;
    }
}

exports.NewAdapter = (parameters) => new Seek(parameters);

File: pkg/siteadapters/seek.go
package siteadapters

import (
	"strings"

	"github.com/JYGC/SkillSurvey/pkg/config"
)

const seekConfigPath = "./pkg/siteadapters/Seek.json"

type Seek struct {
	SiteAdapterBase
}

func NewSeek() *Seek {
	seekAdapter := new(Seek)
	config.JsonToConfig(&seekAdapter.ConfigSettings, seekConfigPath)
	seekAdapter.SiteName = "Seek.com.au"
	seekAdapter.JobPostLink = "[data-automation=\"jobTitle\"]"
	seekAdapter.TitleSelector = "[data-automation=\"job-detail-title\"]"
	seekAdapter.BodySelector = "[data-automation=\"jobAdDetails\"]"
	seekAdapter.PostedDateSelector = "span.FYwKg._2Bz3E.C6ZIU_4._6ufcS_4._3KSG8_4._29m7__4._2WTa0_4"
	seekAdapter.CitySelector = "div.FYwKg._3VxpE_4 > div:nth-child(1)"
	seekAdapter.Country = "Australia"
	seekAdapter.SuburbSelector = "div.FYwKg._3VxpE_4 > div:nth-child(2)"
	seekAdapter.TitleType = "text"
	seekAdapter.BodyType = seekAdapter.TitleType
	seekAdapter.PostedDateType = seekAdapter.TitleType
	seekAdapter.CityType = seekAdapter.TitleType
	seekAdapter.SuburbType = seekAdapter.TitleType
	return seekAdapter
}

func (s Seek) GetJobSiteNumber(url string, doc string) string {
	return url[strings.Index(url, "/job/")+5 : strings.LastIndex(url, "?")]
}

func (s Seek) GetPostedDate(url string, doc string) string {
	return ""
}
